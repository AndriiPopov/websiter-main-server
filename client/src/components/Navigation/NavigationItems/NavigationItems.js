import React from 'react';
import classes from './NavigationItems.module.css';
import NavigationItem from './NavigationItem/NavigationItem';

const navigationItems = () => (
    <ul className={classes.NavigationItems}>
        <NavigationItem 
            link="/dashboard"
            icon='<svg width="24" height="24" viewBox="0 0 24 24"><path d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"></path></svg>'>
            Dashboard
        </NavigationItem>
        <NavigationItem 
            link="/editor"
            icon='<svg width="24" height="24" viewBox="0 0 24 24"><path d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"></path></svg>'>
            Editor
        </NavigationItem>
        <NavigationItem 
            link="/settings"
            icon='<svg width="24" height="24" viewBox="0 0 24 24"><path d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"></path></svg>'>
            Settings
        </NavigationItem>
        <NavigationItem 
            link="/user"
            icon='<svg width="24" height="24" viewBox="0 0 24 24"><path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"></path></svg>'>
            User settings
        </NavigationItem>
        <NavigationItem 
            link="/logout"
            icon='<svg width="24" height="24" viewBox="0 0 24 24"><path d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"></path></svg>'>
            Logout
        </NavigationItem>
    </ul>
);

export default navigationItems;