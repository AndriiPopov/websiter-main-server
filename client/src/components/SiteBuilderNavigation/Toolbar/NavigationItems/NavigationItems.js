import React from 'react';
import { connect } from 'react-redux';

import classes from './NavigationItems.module.css';
import * as actions from '../../../../store/actions/index';
import NavigationItem from './NavigationItem/NavigationItem';
import UndoRedo from './UndoRedo/UndoRedo';
import NavigationItemButtonDropDown from './NavigationItemButtonDropDown/NavigationItemButtonDropDown';
import NavigationItemButton from './NavigationItemButton/NavigationItemButton';
import Aux from '../../../../hoc/Aux';
import PagesMenu from '../ToolsSet/DropDownMenus/PagesMenu/PagesMenu';
import BuilderZoomMenu from '../ToolsSet/DropDownMenus/BuilderZoomMenu/BuilderZoomMenu';

export const NavigationItems = props => (
    <ul className={classes.NavigationItems}>
        { !props.previewIsOn ?
                <Aux>
                    <NavigationItem type='sections'>Sections</NavigationItem>
                    <NavigationItem type='boxes'>Boxes</NavigationItem>
                    <NavigationItem type='text'>Text</NavigationItem>
                </Aux>
            : null
        }

        <ul className={classes.NavigationItemsRight}>
            <NavigationItemButtonDropDown
                icon='<svg width="24" height="24" viewBox="0 0 24 24"><path d="M1 5h2v14H1zm4 0h2v14H5zm17 0H10c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1zM11 17l2.5-3.15L15.29 16l2.5-3.22L21 17H11z"></path></svg>'
                width="500">
                <PagesMenu />
            </NavigationItemButtonDropDown>
            <NavigationItemButton
                active={props.previewIsOn} 
                clicked={() => props.previewClicked()}
                icon='<svg width="24" height="24" viewBox="0 0 24 24"><path d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"></path></svg>' />
            <NavigationItemButtonDropDown 
                icon='<svg width="24" height="24" viewBox="0 0 24 24"><path d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"></path></svg>'>
                <BuilderZoomMenu />
            </NavigationItemButtonDropDown>
                
            { !props.previewIsOn ? <UndoRedo /> : null }
        </ul>
    </ul>
);

const mapStateToProps = state => {
    return {
        previewIsOn: state.builder.present.preview
    }
};

const mapDispatchToProps = dispatch => {
    return {
        previewClicked: () => dispatch(actions.previewPage())
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(NavigationItems);