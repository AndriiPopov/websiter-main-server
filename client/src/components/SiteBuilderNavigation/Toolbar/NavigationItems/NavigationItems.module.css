.NavigationItems {
    margin: 0px;
    padding: 0px;
    list-style: none;
    display: flex;
    align-items: center;
    height: 30px;
    flex-flow: row;
    width: 100%;
}

.NavigationItemsRight {
    margin: 0px;
    padding: 0px;
    list-style: none;
    display: flex;
    align-items: center;
    height: 100%;
    flex-flow: row;
    float: right;
    margin-left: auto;
}

