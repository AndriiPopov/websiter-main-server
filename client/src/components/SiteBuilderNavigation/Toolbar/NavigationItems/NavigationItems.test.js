import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { NavigationItems } from './NavigationItems';

configure({adapter: new Adapter()});

describe('NavigationItems', () => {
    it('should be defined', () => {
        expect(NavigationItems).toBeDefined();
    });

    it('it should render button with onclick null when disabled', () => {
        const tree = shallow(
            <NavigationItems />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});