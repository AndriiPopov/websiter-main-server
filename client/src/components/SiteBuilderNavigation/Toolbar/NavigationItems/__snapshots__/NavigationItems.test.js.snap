// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItems it should render button with onclick null when disabled 1`] = `
<aux>
  <ul
    className="NavigationItems"
  >
    <Connect(NavigationItem)
      type="sections"
    >
      Sections
    </Connect(NavigationItem)>
    <Connect(NavigationItem)
      type="boxes"
    >
      Boxes
    </Connect(NavigationItem)>
    <ul
      className="NavigationItemsRight"
    >
      <NavigationItemButtonDropDown
        dropDownType="pageZoom"
        icon="<svg height=\\"20\\" viewBox=\\"0 0 24 24\\" width=\\"20\\"><path d=\\"M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z\\"></path></svg>"
      />
      <Connect(UndoRedo) />
    </ul>
  </ul>
</aux>
`;
